### PR TITLE
Add photo display functionality

### PR DESCRIPTION
**Pull request by freszu**

### CODE REVIEW

File: app/src/main/java/pl/temomuko/autostoprace/data/model/LocationRecord.java [hunk 1]
```diff
@@ -231,23 +231,10 @@ public LocationRecord createFromParcel(Parcel source) {
...
-            if (in.peek() == JsonToken.NULL) {
-                in.nextNull();
-                return null;
-            } else {
-                return Uri.parse(IMAGE_URL_PREFIX + in.nextString());
-            }
+    @Override
+    public void postProcessGson() {
+        if (mImageLocation != null) {
+            mImageLocation = Constants.API_BASE_URL + "uploads/location/image/" + mId + "/" + mImageLocation;
```
Comment (szymonkozak) [inline app/src/main/java/pl/temomuko/autostoprace/data/model/LocationRecord.java]: Why did you remove IMAGE_URL_PREFIX and use harcoded url string?

File: app/src/main/java/pl/temomuko/autostoprace/ui/teamslocationsmap/ClasterUtil.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+
+import android.support.annotation.NonNull;
+
+import java.util.Collection;
+import java.util.Iterator;
+
+import pl.temomuko.autostoprace.ui.teamslocationsmap.adapter.map.LocationRecordClusterItem;
+
+public class ClasterUtil {
```
Comment (szymonkozak) [inline app/src/main/java/pl/temomuko/autostoprace/ui/teamslocationsmap/ClasterUtil.java]: final

File: app/src/main/java/pl/temomuko/autostoprace/ui/teamslocationsmap/adapter/map/LocationRecordClusterRenderer.java [hunk 3]
```diff
@@ -9,51 +12,69 @@
...
- * Created by Rafał Naniewicz on 02.04.2016.
- */
 public class LocationRecordClusterRenderer extends DefaultClusterRenderer<LocationRecordClusterItem> {
 
     private static final int MIN_CLUSTER_SIZE = 10;
-    private static final String TAG = LocationRecordClusterRenderer.class.getSimpleName();
 
     private final Context mContext;
+    private final String emptyTitleString;
+    private final String pressToSeeThePhotoString;
```
Comment (szymonkozak) [inline app/src/main/java/pl/temomuko/autostoprace/ui/teamslocationsmap/adapter/map/LocationRecordClusterRenderer.java]: "String" suffix is unnecesary, I know type of this field, maybe you should use emptyTitle and pressToSeeThePhotoMessage

File: app/src/main/java/pl/temomuko/autostoprace/ui/teamslocationsmap/adapter/map/LocationRecordClusterRenderer.java [hunk 4]
```diff
@@ -9,51 +12,69 @@
...
+                markerOptions.title(mContext.getString(R.string.msg_last_location_record_message) + "\n"
+                        + lastTitle);
+            }
+
+            if (lastImageUri == null) {
+                markerOptions.snippet(lastSnippet);
+            } else {
+                markerOptions.snippet(getMarkerWithPhotoSnippet(lastSnippet));
+            }
+
```
Comment (szymonkozak) [inline app/src/main/java/pl/temomuko/autostoprace/ui/teamslocationsmap/adapter/map/LocationRecordClusterRenderer.java]: In my opinion you should extract setting title and snippet to 2 other methods.

File: app/src/main/java/pl/temomuko/autostoprace/ui/widget/FullScreenImageDialog.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+        View view = inflater.inflate(R.layout.full_screen_image_dialog, container, false);
+        ButterKnife.bind(this, view);
+        Uri imageUri = getArguments().getParcelable(EXTRA_IMAGE_URI);
+        showImage(imageUri);
+
+        setupListeners();
+
+        return view;
+    }
```
Comment (szymonkozak) [inline app/src/main/java/pl/temomuko/autostoprace/ui/widget/FullScreenImageDialog.java]: Unnecessary enters.

File: app/src/main/java/pl/temomuko/autostoprace/ui/widget/FullScreenImageDialog.java [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+        mMaterialProgressBar.setVisibility(View.GONE);
+        Toast.makeText(getContext(), R.string.error_while_loading_photo, Toast.LENGTH_LONG).show();
+    }
+
+    private void setupListeners() {
+        mCloseButton.setOnClickListener(view -> dismiss());
+    }
+
+
+}
```
Comment (szymonkozak) [inline app/src/main/java/pl/temomuko/autostoprace/ui/widget/FullScreenImageDialog.java]: Too many enters.

File: app/src/main/res/layout/full_screen_image_dialog.xml [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+        android:layout_height="match_parent"
+        />
+
+    <me.zhanghai.android.materialprogressbar.MaterialProgressBar
+        android:id="@+id/mpb_loading_progress"
+        android:layout_width="60dp"
+        android:layout_height="60dp"
+        android:layout_gravity="center"
+        app:mpb_progressStyle="circular"
+        />
```
Comment (szymonkozak) [inline app/src/main/res/layout/full_screen_image_dialog.xml]: We use these dimensions (60dp) of progress bar in 3 places, this should go to dimens.xml (or styles.xml if you want to move all progress bar properties to one place)

File: app/src/main/res/layout/full_screen_image_dialog.xml [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+    <ImageView
+        android:id="@+id/close_button"
```
Comment (szymonkozak) [inline app/src/main/res/layout/full_screen_image_dialog.xml]: btn_close, not close_button

File: app/src/main/res/layout/full_screen_image_dialog.xml [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_gravity="top|right"
+        android:padding="@dimen/default_margin"
+        android:src="@drawable/remove_photo_button"/>
```
Comment (szymonkozak) [inline app/src/main/res/layout/full_screen_image_dialog.xml]: same as above

File: app/src/main/res/values/styles.xml [hunk 10]
```diff
@@ -103,4 +103,8 @@
         <item name="android:textSize">@dimen/bottom_sheet_text_size</item>
     </style>
 
+    <style name="full_screen_dialog" parent="android:Theme.Black.NoTitleBar.Fullscreen">
+        <item name="android:windowAnimationStyle">@android:style/Animation.Dialog</item>
+    </style>
+
```
Comment (szymonkozak) [inline app/src/main/res/values/styles.xml:111]: There is multiple conventions of naming styles in this file, we should standardize this.

File: app/src/main/res/layout/full_screen_image_dialog.xml [hunk 11]
```diff
@@ -0,0 +1,30 @@
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+             xmlns:app="http://schemas.android.com/apk/res-auto"
+             android:layout_width="match_parent"
+             android:layout_height="match_parent"
+             android:background="@android:color/black"
+             android:orientation="vertical">
+
+    <com.davemorrissey.labs.subscaleview.SubsamplingScaleImageView
+        android:id="@+id/subsampling_image_view"
```
Comment (szymonkozak) [inline app/src/main/res/layout/full_screen_image_dialog.xml]: iv_subsampling, not subsampling_image_view

File: app/src/main/java/pl/temomuko/autostoprace/ui/widget/FullScreenImageDialog.java [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+
+public class FullScreenImageDialog extends DialogFragment {
+
+    public static final String TAG = FullScreenImageDialog.class.getSimpleName();
+
+    @BindView(R.id.close_button) ImageView mCloseButton;
+    @BindView(R.id.subsampling_image_view) SubsamplingScaleImageView mSubsamplingScaleImageView;
+    @BindView(R.id.mpb_loading_progress) MaterialProgressBar mMaterialProgressBar;
+
+    public static final String EXTRA_IMAGE_URI = "extra_image_uri";
```
Comment (szymonkozak) [inline app/src/main/java/pl/temomuko/autostoprace/ui/widget/FullScreenImageDialog.java]: private

**Review by szymonkozak — CHANGES_REQUESTED** [review]

File: app/src/main/java/pl/temomuko/autostoprace/data/model/LocationRecord.java (hunk 1 above)
Comment (freszu) [reply app/src/main/java/pl/temomuko/autostoprace/data/model/LocationRecord.java]: done

File: app/src/main/java/pl/temomuko/autostoprace/ui/teamslocationsmap/ClasterUtil.java (hunk 2 above)
Comment (freszu) [reply app/src/main/java/pl/temomuko/autostoprace/ui/teamslocationsmap/ClasterUtil.java]: done

File: app/src/main/java/pl/temomuko/autostoprace/ui/teamslocationsmap/adapter/map/LocationRecordClusterRenderer.java (hunk 3 above)
Comment (freszu) [reply app/src/main/java/pl/temomuko/autostoprace/ui/teamslocationsmap/adapter/map/LocationRecordClusterRenderer.java]: done

File: app/src/main/java/pl/temomuko/autostoprace/ui/widget/FullScreenImageDialog.java (hunk 12 above)
Comment (freszu) [reply app/src/main/java/pl/temomuko/autostoprace/ui/widget/FullScreenImageDialog.java]: done

File: app/src/main/java/pl/temomuko/autostoprace/ui/widget/FullScreenImageDialog.java (hunk 6 above)
Comment (freszu) [reply app/src/main/java/pl/temomuko/autostoprace/ui/widget/FullScreenImageDialog.java]: done

File: app/src/main/res/layout/full_screen_image_dialog.xml (hunk 11 above)
Comment (freszu) [reply app/src/main/res/layout/full_screen_image_dialog.xml]: done

File: app/src/main/res/layout/full_screen_image_dialog.xml (hunk 7 above)
Comment (freszu) [reply app/src/main/res/layout/full_screen_image_dialog.xml]: done

File: app/src/main/res/layout/full_screen_image_dialog.xml (hunk 8 above)
Comment (freszu) [reply app/src/main/res/layout/full_screen_image_dialog.xml]: done

File: app/src/main/res/layout/full_screen_image_dialog.xml (hunk 9 above)
Comment (freszu) [inline app/src/main/res/layout/full_screen_image_dialog.xml]: not an id

Comment (freszu) [reply app/src/main/res/layout/full_screen_image_dialog.xml]: not sure if thats what you meant but changed name of drawable to better one

File: app/src/main/res/values/styles.xml (hunk 10 above)
Comment (freszu) [reply app/src/main/res/values/styles.xml:111]: inserted fixme comment

File: app/src/main/res/layout/full_screen_image_dialog.xml (hunk 9 above)
Comment (szymonkozak) [reply app/src/main/res/layout/full_screen_image_dialog.xml]: Ok, nevermind, I didn't notice that is drawable, not id of view,

**Review by szymonkozak — APPROVED** [review]
